python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -392,29 +395,35 @@ def get_next_partition_key(
             return start_time.strftime(self.fmt)
 
     def get_next_partition_window(
-        self, end_dt: datetime, current_time: Optional[datetime] = None
+        self, end_dt: datetime, current_time: Optional[datetime] = None, respect_bounds: bool = True
     ) -> Optional[TimeWindow]:
-        last_partition_window = self.get_last_partition_window(current_time)
-        if last_partition_window is None:
-            return None
-
         windows_iter = iter(self._iterate_time_windows(end_dt))
         next_window = next(windows_iter)
-        if next_window.start >= last_partition_window.end:
-            return None
-        else:
-            return next_window
 
-    def get_prev_partition_window(self, start_dt: datetime) -> Optional[TimeWindow]:
+        if respect_bounds:
+            last_partition_window = self.get_last_partition_window(current_time)
+            if last_partition_window is None:
+                return None
+
+            if next_window.start >= last_partition_window.end:
+                return None
+
+        return next_window
+
+    def get_prev_partition_window(
+        self, start_dt: datetime, respect_bounds: bool = True
+    ) -> Optional[TimeWindow]:
         windows_iter = iter(self._reverse_iterate_time_windows(start_dt))
         prev_window = next(windows_iter)
         first_partition_window = self.get_first_partition_window()
-        if first_partition_window is None or prev_window.start < first_partition_window.start:
+        if respect_bounds and (
+            first_partition_window is None or prev_window.start < first_partition_window.start
+        ):
             return None
         else:
             return prev_window
 
-    @functools.lru_cache(maxsize=5)
+    @functools.lru_cache(maxsize=256)
```

Comment:
> so this should basically just be set to around the number of distinct TimeWindowPartitionsDefinitions we expect to see per asset graph if I'm understanding correctly.

Some users have many different `start_date`s, so there could be a bunch of distinct `TimeWindowPartitionsDefinition`s. I thought it would make sense to err on the side of larger, because these objects are pretty small so unlikely to have much impact on the process's memory consumption, and the negative impact of paging back and forth could be high.